src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,37 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newFixedThreadPool(5);
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        try {
+            if (!executorService.awaitTermination(1, TimeUnit.SECONDS)) {
+                executorService.shutdown();
+            }
```

Comment:
The shutdown method should be calling shutdownNow instead of shutdown if the awaitTermination returns false. The shutdown method initiates an orderly shutdown in which previously submitted tasks are executed, but no new tasks will be accepted. In contrast, shutdownNow attempts to stop all actively executing tasks and halts the processing of waiting tasks.